// pg-styleguide-assets/components/tabs/tab-item.jsx
PG.Components.TabItem = React.createClass({
  propTypes: {
    text: React.PropTypes.string,
    index: React.PropTypes.number,
    active: React.PropTypes.bool,
    badge: React.PropTypes.string,
    clickHandler: React.PropTypes.func
  },

  getDefaultProps: function getDefaultProps() {
    return {
      text: "Tab Item"
    };
  },

  handleClick: function(e) {
    e.preventDefault();
    e.stopPropagation();

    if (this.props.clickHandler)
      this.props.clickHandler(this.props.index);
  },

  render: function render() {
    var classes = React.addons.classSet({
      "active" : this.props.active
    });

    return (
      <li className={classes}>
        <a href="#" onClick={this.handleClick}>{this.props.text}
        {this.getBadge()}
        </a>
      </li>
    );
  },

  getBadge: function getBadge() {
    if (!this.props.badge) return;

    return (
      <span className="badge">{this.props.badge}</span>
    );
  }
});

if(typeof module !== "undefined") module.exports = PG.Components.TabItem;
