// pg-styleguide-assets/components/tabs/tabs.jsx
PG.Components.Tabs = React.createClass({
  propTypes: {
    items: React.PropTypes.array,
    activeIndex: React.PropTypes.number,
    pills: React.PropTypes.bool
  },

  getDefaultProps: function getDefaultProps() {
    return {
      items: [],
      activeIndex: 0
    };
  },

  getInitialState: function() {
    return {
      activeIndex: this.props.activeIndex
    };
  },

  componentWillReceiveProps: function(nextProps) {
    if (nextProps.activeIndex)
      this.setState({
        activeIndex: nextProps.activeIndex
      });
  },

  handleClick: function(index) {
    if (this.props.clickHandler)
      this.props.clickHandler(index);

    this.setState({
      activeIndex: index
    });
  },

  render: function() {
    var classes = React.addons.classSet({
      "tabs" : true,
      "pills" : this.props.pills
    });

    return (
      <ul className={classes}>
      {this.props.items.map(function(item, index){
          return (
            <PG.Components.TabItem
              key={index}
              text={item.text}
              index={index}
              active={this.state.activeIndex === index}
              badge={item.badge}
              clickHandler={this.handleClick}/>
          )}.bind(this)
      )}
        <li tabIndex="-1"></li>
      </ul>
    );
  }
});

if(typeof module !== "undefined") module.exports = PG.Components.Tabs;
