// pg-styleguide-assets/components/navpanel/navigation-item.jsx
PG.Components.NavigationItem = React.createClass({
  propTypes: {
    text: React.PropTypes.string,
    warning: React.PropTypes.bool,
    count: React.PropTypes.number,
    icon: React.PropTypes.string
  },

  getDefaultProps: function getDefaultProps() {
    return {
      text: "Navigation Item"
    };
  },

  handleClick: function handleClick(e) {
    e.preventDefault();
    e.stopPropagation();

    if (this.props.clickHandler) this.props.clickHandler(this);
  },

  render: function render() {
    return (
      <li>
        <a href={this.props.url} onClick={this.handleClick}>
          {this.getIcon()}
          <span className="label">{this.props.text}</span>
          {this.getCount()}
        </a>
      </li>
    );
  },

  getIcon: function getIcon() {
    if (!this.props.icon) return;

    var classes = "icon-" + this.props.icon;
    return (
      <span className={classes}></span>
    );
  },

  getCount: function getCount() {
    if (!this.props.count) return;

    var classes = React.addons.classSet({
      "badge": true,
      "warning": this.props.warning
    });

    return (
      <span className="badge-wrapper">
        <span className={classes}>{this.props.count}</span>
      </span>
    );
  }
});

if(typeof module !== "undefined") module.exports = PG.Components.NavigationItem;
