// pg-styleguide-assets/components/dropdown/menu.jsx
PG.Components.DropdownMenu = React.createClass({
  propTypes:  {
    items:          React.PropTypes.array,
    open:           React.PropTypes.bool,
    transparent:    React.PropTypes.bool,
    textCentered:   React.PropTypes.bool,
    bold:           React.PropTypes.bool,
    selectHandler:  React.PropTypes.func
  },

  getDefaultProps: function getDefaultProps() {
    return {
      items: [],
      open: false,
      transparent: false,
      textCentered: false,
      updateText: false
    };
  },

  render: function render() {
    var items, classes;

    classes = React.addons.classSet({
      "dropdown-menu" : true,
      "open"          : this.props.open,
      "transparent"   : this.props.transparent,
      "text-centered" : this.props.textCentered,
      "bold"          : this.props.bold
    });

    items = this.props.items.map(function(item, index){
      return (
        <PG.Components.DropdownMenuItem
          key={index}
          value={item.value}
          text={item.text}
          type={item.type}
          selectHandler={this.props.selectHandler} />
      );
    }.bind(this));

    return (
      <ul className={classes}>{items}</ul>
    );
  }
});

if(typeof module !== "undefined") module.exports = PG.Components.DropdownMenu;
