// pg-styleguide-assets/components/dropdown/menu-item.jsx
PG.Components.DropdownMenuItem = React.createClass({
  propTypes:  {
    type:            React.PropTypes.string,
    value:           React.PropTypes.string,
    text:            React.PropTypes.string,
    selectHandler:   React.PropTypes.func
  },

  getDefaultProps: function getDefaultProps() {
    return {
      type: "item",
      text: "Menu Item"
    };
  },

  handleClick: function handleClick(el) {
    if (this.props.selectHandler)
      this.props.selectHandler(el);
  },

  render: function() {
    var classes;

    classes = React.addons.classSet({
      "item"    : this.props.type === "item",
      "title"   : this.props.type === "title",
      "divider" : this.props.type === "divider"
    });

    return (
      <li className={classes}>
        {this.getItem()}
      </li>
    );
  },

  getItem: function getItem() {
    if (this.props.type === "divider") return;

    if (this.props.type === "title") {
      return <span>{this.props.text}</span>;
    } else {
      return (
        <label forHTML={this.props.value} >

          <input type="radio"
            onClick={this.handleClick}
            name={this.props.value}
            value={this.props.value}
            data-text={this.props.text} />
          {this.props.text}

        </label>
      );
    }
  }
});

if(typeof module !== "undefined") module.exports = PG.Components.DropdownMenuItem;
