// pg-styleguide-assets/components/paginator/paginator-item.jsx
PG.Components.PaginatorItem = React.createClass({
  propTypes:  {
    pageIndex:      React.PropTypes.oneOfType([
      React.PropTypes.string,
      React.PropTypes.number
    ]),
    type:           React.PropTypes.string,
    text:           React.PropTypes.string,
    disabled:       React.PropTypes.bool,
    clickHandler:   React.PropTypes.func
  },

  getDefaultProps: function getDefaultProps() {
    return {
      type: "page"
    };
  },

  handleClick: function handleClick(e) {
    e.preventDefault();
    e.stopPropagation();

    if (this.props.clickHandler && !this.props.disabled)
      this.props.clickHandler(this.props.pageIndex);
  },

  render: function render() {
    var classes = React.addons.classSet({
      "page"    : this.props.type === "page",
      "prev"    : this.props.type === "prev",
      "next"    : this.props.type === "next",
      "current"  : this.props.current
    });

    return (
      <li className={classes} onClick={this.handleClick} disabled={this.props.disabled}>
        <a href="#">{this.props.text}</a>
      </li>
    );
  }
});

if(typeof module !== "undefined") module.exports = PG.Components.PaginatorItem;
