// pg-styleguide-assets/components/label/label.jsx
PG.Components.Label = React.createClass({
  allowedSizes: [ "small", "large"],
  allowedRoles: [ "primary", "secondary", "additive", "destructive", "grey", "orange", "purple"],

  propTypes:  {
    size:           React.PropTypes.string,
    role:           React.PropTypes.string,
    text:           React.PropTypes.string
  },

  getDefaultProps: function getDefaultProps() {
    return {
      size: "large",
      role: "primary",
      text: "Label"
    };
  },

  render: function render() {
    var classSet = this.getClassSet(),
        classes = React.addons.classSet(classSet);

    return (
      <span className={classes}>{this.props.text}</span>
    );
  },

  getClassSet: function getClassSet() {
    var classSet = {
      "label"   : true,
      "primary" : this.allowedRoles.indexOf(this.props.role) === -1,
      "large"   : this.allowedSizes.indexOf(this.props.size) === -1
    };

    classSet[this.props.size] = this.allowedSizes.indexOf(this.props.size) > -1;
    classSet[this.props.role] = this.allowedRoles.indexOf(this.props.role) > -1;

    return classSet;
  }
});

if(typeof module !== "undefined") module.exports = PG.Components.Label;
