// pg-styleguide-assets/components/breadcrumb/breadcrumb.jsx
PG.Components.Breadcrumb = React.createClass({
  propTypes: {
    items: React.PropTypes.array,
    clickHandler: React.PropTypes.func,
    delimiterIcon: React.PropTypes.string,
    header: React.PropTypes.bool
  },

  getDefaultProps: function getDefaultProps() {
    return {
      items: []
    };
  },

  render: function render() {
    var classes, items;

    classes = React.addons.classSet({
      "breadcrumbs" : true,
      "inline" : !this.props.header
    });

    items = this.props.items.map(function(item, index) {
      return (
        <PG.Components.BreadcrumbItem
          key={index}
          url={item.url}
          text={item.text}
          disabled={index === this.props.items.length - 1}
          clickHandler={this.props.clickHandler}
          icon={item.icon}
          delimiterIcon={this.props.delimiterIcon}
        />
      );
    }.bind(this));

    return (
      <ul className={classes}>{items}</ul>
    );
  }
});

if(typeof module !== "undefined") module.exports = PG.Components.Breadcrumb;
