// pg-styleguide-assets/components/breadcrumb/breadcrumb-item.jsx
PG.Components.BreadcrumbItem = React.createClass({
  propTypes: {
    text: React.PropTypes.string,
    url: React.PropTypes.string,
    disabled: React.PropTypes.bool,
    clickHandler: React.PropTypes.func,
    delimiterIcon: React.PropTypes.string
  },

  getDefaultProps: function getDefaultProps() {
    return {
      text: "Breadcrumb Item",
      url: "#"
    };
  },

  handleClick: function handleClick(e) {
    e.preventDefault();
    e.stopPropagation();

    if (this.props.clickHandler && !this.props.disabled)
      this.props.clickHandler(this);
  },

  render: function render() {
    return (
      <li>
        <a href={this.props.url} onClick={this.handleClick} disabled={this.props.disabled}>
          {this.getIcon()}
          {this.props.text}
          {this.getDelimiter()}
        </a>
      </li>
    );
  },

  getIcon: function getIcon() {
    if (!this.props.icon) return;

    var classes = "icon icon-" + this.props.icon;
    return (<span className={classes}></span>);
  },

  getDelimiter: function getDelimiter() {
    if (this.props.disabled || !this.props.delimiterIcon) return;

    var classes = "icon icon-" + this.props.delimiterIcon;
    return (<span className={classes}></span>);
  }
});

if(typeof module !== "undefined") module.exports = PG.Components.BreadcrumbItem;
